custom_components/econnect_metronet/config_flow.py

```diff
@@ -127,28 +130,30 @@ async def async_step_init(self, user_input=None):
 
         # Populate with latest changes or previous settings
         user_input = user_input or {}
-        suggest_arm_home = user_input.get(CONF_AREAS_ARM_HOME) or self.config_entry.options.get(CONF_AREAS_ARM_HOME)
-        suggest_arm_night = user_input.get(CONF_AREAS_ARM_NIGHT) or self.config_entry.options.get(CONF_AREAS_ARM_NIGHT)
-        suggest_arm_vacation = user_input.get(CONF_AREAS_ARM_VACATION) or self.config_entry.options.get(
-            CONF_AREAS_ARM_VACATION
-        )
         suggest_scan_interval = user_input.get(CONF_SCAN_INTERVAL) or self.config_entry.options.get(CONF_SCAN_INTERVAL)
+
+        # Generate sectors list for user config options
+        sectors_list = []
+        device = self.hass.data[DOMAIN][self.config_entry.entry_id][KEY_DEVICE]
+        for sector_id, item in device._inventory.get(q.SECTORS, {}).items():
+            sectors_list.append(f"{item['element']} : {item['name']}")
```

Comment:
did you try to pass a dictionary like:
```python
{3: "Sector 1", 4: "Sector 2"}
```

and so on?

Reply:
In reality, I can answer both questions with the same response. 
The setting was my choice because we need the "element" to perform the 'get_state' function

https://github.com/palazzem/ha-econnect-alarm/blob/5f4d87b8596efc81b9b744d5d4b3c7e38acb0d6c/custom_components/econnect_metronet/devices.py#L182

Reply:
And for the dictionary, I have modified the method of the query, and in the next commit, I have inserted a dictionary